test/functional/package/list_test.rb

```diff
@@ -0,0 +1,77 @@
+require 'test_helper'
+require 'functional/test_helper'
+
+describe 'listing packages' do
+  before do
+    @cmd = %w(package list)
+  end
+
+  let(:org_id) { 1 }
+  let(:product_id) { 1 }
+  let(:repo_id) { 1 }
+  let(:empty_response) do
+    {
+      "total" => 0,
+      "subtotal" => 0,
+      "page" => "1",
+      "per_page" => "1000",
+      "error" => nil,
+      "search" => nil,
+      "sort" => {
+        "by" => nil,
+        "order" => nil
+      },
+      "results" => []
+    }
+  end
+
+  it "lists packages for an organization" do
+    params = ["--organization-id=#{org_id}"]
+
+    ex = api_expects(:packages, :index, 'Packages list') do |par|
+      par['organization_id'] == org_id &&
+        par['page'] == 1 &&
+        par['per_page'] == 1000
+    end
+
+    ex.returns(empty_response)
+    expected_result = [
+      '---|----------|-----------',
+      'ID | FILENAME | SOURCE RPM',
+      '---|----------|-----------',
+      ''
+    ]
+
+    expected_result = success_result(expected_result.join("\n"))
+
+    result = run_cmd(@cmd + params)
+    assert_cmd(expected_result, result)
+  end
+
+  it "lists packages for a repository" do
+    params = [
+      "--product-id=#{product_id}",
+      "--repository-id=#{repo_id}"
+    ]
+
+    ex = api_expects(:packages, :index, 'repository packages list') do |par|
+      par['product_id'] == product_id &&
```

Comment:
`product_id` isn't being passed to `Packages#index`; I don't see in the API where it is a param for [this endpoint](http://www.katello.org/docs/3.0/api/apidoc/packages.html).  Also, is there a reason you are validating the default page and per_page options? I believe we have [pagination tests](https://github.com/theforeman/hammer-cli-foreman/blob/master/test/functional/commands/list_test.rb#L4) that ensure those parameters are properly passed.